flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -2411,6 +2430,7 @@ class ArrayExprLowering {
     }
     if (useEmbox) {
       auto boxTy = fir::BoxType::get(reduceRank(arrTy, slice));
+      // TODO: fir.rebox when the memref is a fir.box.
```

Comment:
Sounds like this one could be a `if (blah.isa<box>()) TODO`.

The trouble with TODO comments, of course, being that it gives users the false sense that something may have been intended to work correctly, when the reality is the opposite.

Reply:
Sure, I added one (current code was creating an mlir verification error anyway, but a TODO gives more feedback that we are in control).